pkg/api/types.go

```diff
@@ -3393,7 +3393,8 @@ type FinalizerName string
 // These are internal finalizer values to Kubernetes, must be qualified name unless defined here or
 // in metav1.
 const (
-	FinalizerKubernetes FinalizerName = "kubernetes"
+	FinalizerKubernetes        FinalizerName = "kubernetes"
+	FinalizerServiceController               = "service-controler"
```

Comment:
Should I use the `kubernetes` one? Felt kinda weird but I couldn't find any other examples since that could potentially conflict with other controllers over time.